aws/resource_aws_ecs_service.go

```diff
@@ -1043,6 +1053,12 @@ func resourceAwsEcsServiceUpdate(d *schema.ResourceData, meta interface{}) error
 		}
 	}
 
+	if d.Get("wait_for_steady_state").(bool) {
+		if err := resourceAwsEcsWaitForServiceSteadyState(d, meta); err != nil {
```

Comment:
This method definitely follows the construct used across the provider when implementing custom `StateChangeConf` functionality, but in this case since the AWS Go SDK provides this method with `ECS.WaitUntilServicesStable` and `ECS.WaitUntilServicesStableWithContext` to add configurable waiter options, I would also suggest adopting the SDK's method in place of this custom logic

Reply:
Ah, fair enough -- I don't think I noticed the `WithContext` variants. Agree this seems like the right approach.